### DISCUSSION TITLE
Link error for MsQuic+OpenSSL static build

### DISCUSSION THREAD

**Discussion by Wizmann** [microsoft/msquic#2142] [Q&A] [answered]
We come across a link error when we want to integrate MsQuic static library with our code, on Win10 21H1 with VS2019

The `msquic.lib` was successfully built with command like this: `scripts/build.ps1 -Config Release -Arch x64 -Static -Tls openssl -Clean`

But when we're trying to it with our code, it says: `warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs;`

I thought it should not be a problem because we make that lib as `/NODEFAULTLIB` [here][1]. But it doesn't work as expected.

Could you please help to figure out the problem? Thanks!

[1]: https://github.com/microsoft/msquic/blob/cd8e64d1bacfa4925427a6599369136dd9ac7f1b/src/bin/CMakeLists.txt#L153

**Comment by nibanks** [comment]
It's interesting that you're seeing an error here. Our CI builds this configuration every time and it doesn't seem to be having any issues (latest run [here](https://dev.azure.com/ms/msquic/_build/results?buildId=241597&view=logs&j=f273a8b6-07fb-5cb9-d2a8-7daab3c38e98&t=63d9d423-841f-53c2-1e8f-aa85cf65a385)). I'm wondering if there is some local SDK difference or something that causes the error. cc @thhous-msft ideas?

**Comment by thhous-msft** [comment, marked as answer]
Ah, it looks like that line is wrong, its only removing libcmt on debug builds. I'll fix. Our CI doesn't run into this because we build tools with the static CRT anyway so it doesnt notice.
